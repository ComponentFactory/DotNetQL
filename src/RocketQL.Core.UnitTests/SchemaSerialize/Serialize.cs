namespace RocketQL.Core.UnitTests.SchemaSerialize;

public class Serialize : UnitTestBase
{
    [Theory]
    [InlineData("printer-schema.graphql")]
    public void FileSchemaDefault(string filename)
    {
        var schemaText = File.ReadAllText(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "TestFiles", filename));
        var schema = new Schema();
        schema.Add(schemaText);
        schema.Validate();
        string print = Serialization.SchemaSerialize(schema);
        print.MatchSnapshot();
    }

    [Theory]
    [InlineData("printer-schema.graphql")]
    public void FileSchemaIndent2Spaces(string filename)
    {
        var schemaText = File.ReadAllText(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "TestFiles", filename));
        var schema = new Schema();
        schema.Add(schemaText);
        schema.Validate();
        string print = Serialization.SchemaSerialize(schema, new SchemaSerializeOptions() { IndentCount = 2 });
        print.MatchSnapshot();
    }

    [Theory]
    [InlineData("printer-schema.graphql")]
    public void FileSchemaIndent2Tabs(string filename)
    {
        var schemaText = File.ReadAllText(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "TestFiles", filename));
        var schema = new Schema();
        schema.Add(schemaText);
        schema.Validate();
        string print = Serialization.SchemaSerialize(schema, new SchemaSerializeOptions() { IndentCharacter = PrintIndentCharacter.Tab, IndentCount = 2 });
        print.MatchSnapshot();
    }

    [Theory]
    [InlineData("printer-schema.graphql")]
    public void FileSchemaNoDescriptions(string filename)
    {
        var schemaText = File.ReadAllText(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "TestFiles", filename));
        var schema = new Schema();
        schema.Add(schemaText);
        schema.Validate();
        string print = Serialization.SchemaSerialize(schema, new SchemaSerializeOptions() { IncludeDescription = false });
        print.MatchSnapshot();
    }

    [Theory]
    [InlineData("printer-schema.graphql")]
    public void FileSchemaShowPredefined(string filename)
    {
        var schemaText = File.ReadAllText(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "TestFiles", filename));
        var schema = new Schema();
        schema.Add(schemaText);
        schema.Validate();
        string print = Serialization.SchemaSerialize(schema, new SchemaSerializeOptions() { IncludeBuiltIn = true });
        print.MatchSnapshot();
    }

    [Theory]
    [InlineData("printer-unrooted-schema.graphql")]
    public void FileSchemaExcludeUnrooted(string filename)
    {
        var schemaText = File.ReadAllText(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "TestFiles", filename));
        var schema = new Schema();
        schema.Add(schemaText);
        schema.Validate();
        string print = Serialization.SchemaSerialize(schema, new SchemaSerializeOptions() { IncludeBuiltIn = true, IncludeUnrooted = false });
        print.MatchSnapshot();
    }

    [Theory]
    [InlineData("printer-unrooted-schema.graphql")]
    public void FileSchemaIncludeUnrooted(string filename)
    {
        var schemaText = File.ReadAllText(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "TestFiles", filename));
        var schema = new Schema();
        schema.Add(schemaText);
        schema.Validate();
        string print = Serialization.SchemaSerialize(schema, new SchemaSerializeOptions() { IncludeBuiltIn = true, IncludeUnrooted = true });
        print.MatchSnapshot();
    }
}

