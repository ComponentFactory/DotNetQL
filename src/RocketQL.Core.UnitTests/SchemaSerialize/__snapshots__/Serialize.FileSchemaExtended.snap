schema
{
    query: unref1
    mutation: unref2
}

type unref1
{
    unref1: Int
}

type unref2
{
    unref2: Int
    extend: Int
}

interface unref3
{
    unref3: Int
    extend: Int
}

union unref4 = unref1 | unref2

input unref5
{
    unref7: Int
    extend: Int
}

enum unref6
{
    unref6
    extend
}

scalar unref7 @specifiedBy(url: "url")

